torch/distributed/_tensor/__init__.py

```diff
@@ -20,6 +20,97 @@
 ]
 
 
+def _dtensor_init_helper(init_op, *size, **kwargs) -> DTensor:
+    # get default device mesh if there's nothing specified
+    device_mesh = kwargs.pop("device_mesh", None)
+    if device_mesh is None:
+        device_mesh = get_global_device_mesh()
+
+    # set default placements to replicated if not specified
+    placements = kwargs.pop("placements", None)
+    if placements is None:
+        placements = [Replicate() for _ in range(device_mesh.ndim)]
+
+    # check device_mesh againts placements
+    assert device_mesh.ndim == len(
+        placements
+    ), "mesh dimension does not match the length of placements"
+
+    # normalize the size argument
+    if len(size) == 1 and isinstance(size[0], Sequence):
+        torch_size = size[0]
+    else:
+        torch_size = list(size)
+    torch_size = torch.Size(torch_size)
+    assert kwargs["layout"] == torch.strided, "layout value not supported!"
+    torch_stride = torch._prims_common.make_contiguous_strides_for(torch_size)
+
+    # get local tensor shape
+    local_shape = compute_local_shape(torch_size, device_mesh, placements)
+    # initialize the local tensor
+    if len(local_shape) == 0:
+        local_tensor = torch.empty(0, **kwargs)
+    else:
+        local_tensor = init_op(local_shape, **kwargs)
+
+    return DTensor(
+        local_tensor=local_tensor,
+        device_mesh=device_mesh,
+        placements=placements,
+        shape=torch_size,
+        dtype=local_tensor.dtype,
+        stride=torch_stride,
+        requires_grad=kwargs["requires_grad"],
+    )
+
+
+def ones(
+    *size,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    layout: torch.layout = torch.strided,
```

Comment:
I think we should also add a check in the init help that errors out when the `layout` is not torch.strided, as we don't support sharding a sparse tensor yet.

Reply:
we have it here: https://github.com/pytorch/pytorch/pull/100933/files#diff-587cd2db5ca2137ea89d48d78c63db9c3c7abfaea8d7249b031f7c2b518a7cadR45